btsieve/src/bitcoin/blockchain_info_hex_bitcoin_http_blocksource.rs

```diff
@@ -0,0 +1,169 @@
+        }
+    }
+
+    fn latest_block_without_tx(
+        &self,
+    ) -> impl Future<Item = BlockchainInfoLatestBlock, Error = Error> + Send + 'static {
+        // https://blockchain.info/q/latesthash only works for mainnet, there is no testnet endpoint
+        // we fall-back to [testnet.]blockchain.info/latestblock to retrieve the latest
+        // block hash
+
+        let latest_block_url = format!("{}/latestblock", self.base_url());
+
+        self.client
+            .get(latest_block_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(move |mut response| {
+                response
+                    .json::<BlockchainInfoLatestBlock>()
+                    .map_err(Error::Reqwest)
+            })
+    }
+
+    fn raw_hex_block(
+        &self,
+        block_hash: String,
+        block_height: u32,
+    ) -> impl Future<Item = MinedBlock, Error = Error> + Send + 'static {
+        let raw_block_by_hash_url =
+            format!("{}/rawblock/{}?format=hex", self.base_url(), block_hash);
+
+        self.client
+            .get(raw_block_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(|response_text| hex::decode(response_text).map_err(Error::Hex))
+            .and_then(move |bytes| {
+                deserialize(bytes.as_ref())
+                    .map(|block| {
+                        log::trace!("Got {:?}", block);
+                        MinedBlock::new(block, block_height)
+                    })
+                    .map_err(|e| {
+                        log::error!("Got new block but failed to deserialize it because {:?}", e);
+                        Error::BlockDeserialization(format!(
+                            "Failed to deserialize the response from blockchain.info into a block: {}", e
+                        ))
+                    })
+            })
+    }
+}
+
+impl BlockSource for BlockchainInfoHexHttpBlockSource {
+    type Block = MinedBlock;
+    type Error = Error;
+
+    fn blocks(
+        &self,
+    ) -> Box<dyn Stream<Item = Self::Block, Error = blocksource::Error<Error>> + Send> {
+        // https://www.blockchain.com/api/q (= https://www.blockchain.info/api/q) states:
+        //  "Please limit your queries to a maximum of 1 every 10 seconds." (29/08/2019)
+        //
+        // The Bitcoin blockchain has a mining interval of about 10 minutes.
+        // The poll interval is configured to once every 5 minutes.
+        let poll_interval = match self.network {
+            Network::Mainnet => 300,
+            _ => 0,
+        };
+
+        log::info!(target: "bitcoin::blocksource", "polling for new blocks from blockchain.info on {} every {} seconds", self.network, poll_interval);
+
+        let cloned_self = self.clone();
+
+        let stream = Interval::new_interval(Duration::from_secs(poll_interval))
```

Comment:
The indentation on this method seems a bit weird.
The `or_else` should be on the same level as `map` and `.filter_map` should be on a new line as well.

Reply:
fixed